### ISSUE TITLE
冻结 ZeusAI 启发的完整扩展训练闭环

### ISSUE THREAD

**Issue by titanxxh** [titanxxh/7wd#43]
## Question

依据已关闭的 ZeusAI 迁移、观测封闭搜索、Transformer 表征、资产审计和接口原型决策，冻结新路线的核心训练闭环：新模型如何冷启动，BGA policy warm-start 与监测如何隔离，搜索如何生成 policy target，合法完整终局如何生成 value target，generation、replay window、模型更新、随机流、checkpoint/resume 和失败原子性如何定义；同时明确旧权重、纯基础版、OneStep teacher、hidden-state actor 输入及未经证明的 population 机制不得进入主线。

**Comment by titanxxh** [comment]
## Grilling round 1 — accepted

用户选择 Q1–Q3 全部按推荐，冻结以下根决策：

1. **共同起点 `g0`**：全新 hybrid Transformer 先使用 #44 train partition 做固定预算 policy-only warm-start；BGA 不提供 value。随后在 policy 不接受 search target、OneStep 或旧权重的前提下，用完整基础游戏 + Pantheon + Agora 的 prior-only 合法完整终局自对弈冷启动 actor-relative value，最后封存唯一 `g0`。搜索臂与对照臂从完全相同的 `g0` 分叉。
2. **机制归因双臂**：强制保留等墙钟 Search arm 与 prior-only control。两臂共享 `g0`、模型结构、示范 Policy 锚、优化器、合法终局 value、随机流体系、checkpoint 和评测日程；唯一机制差是 Search arm 使用 observation-closed root visit policy target，control 使用 masked policy action / self-distillation。只比较 Search 与 `g0` 不构成搜索净收益证据。
3. **密封同步 generation**：首轮采用同步、零 staleness 代际。每盘从完整扩展正式初始状态开始并固定同一 generation bundle；只有合法完整终局轨迹可以原子封存，learner 只消费已封存完整代；checkpoint 与 manifest 持久化完成后才能发布下一 bundle。盘中换权重、跨代轨迹、持续异步 actor 和 bounded staleness 全部关闭。

这三项只冻结拓扑和因果边界。示范/自博弈 loss 混合、replay window、每代更新预算、checkpoint/resume、RNG 派生和故障原子性留给后续依赖轮次；历史路线中的“最近四代”“96 generations”或 `rapid-search-v1` 数值不得自动继承。

**Comment by titanxxh** [comment]
## Grilling round 2 — accepted

用户选择 Q4–Q8 全部按推荐，冻结以下依赖决策：

4. **Policy 不变的 Value bootstrap**：policy-only warm-start 后，冻结 tokenizer、Transformer trunk、candidate policy head 与 scatter 所组成的完整 policy 决策函数，只训练 value 专属 adapter/head。value 唯一来自该冻结 prior-only policy 的完整扩展合法终局 actor-relative return；固定 probe 上 bootstrap 前后 policy logits 必须逐值一致。BGA winner、OneStep、search value 和旧模型信号全部禁止。
5. **主 learner 合同**：封存 `g0` 后不携带 warm-start/bootstrap optimizer moments；两个 arm 分别建立字节与配置相同的新 AdamW 状态并解冻全部参数。每批固定 75% 自博弈决策、25% #44 train demonstrations；自博弈 loss 为 masked policy CE + actor-relative terminal value MSE，示范只算 expert masked policy CE。Search target 是完整 legal root visits，control target 是生成时冻结 bundle 的完整 masked soft prior，不是 sampled-action one-hot。无 PPO、entropy bonus、OneStep KL、BGA value 或 monitor 样本。
6. **四代均衡近窗**：生成 `g+1` 时只消费 `g..g-3` 中存在的密封代；先均匀选代、再在代内均匀选 decision。示范锚独立于窗口。禁止 prioritized/outcome/family weighting、跨 arm、跨 seed、旧路线 replay；离开 active window 的 artifact 保留不删。四代是新路线假设，不继承旧 Rapid 证据。
7. **原子恢复**：只保留原子写完且哈希正确的完整游戏；临时/不完整游戏以同一 game index 重生成。generation 在精确索引集完整校验后才 seal；learner 以固定 cadence 和代末 checkpoint 封存模型、optimizer/scaler、计数、逻辑 RNG 位置和 training-mix hashes。resume 只允许同 output root、同全部合同，从最近有效边界继续；小型 interrupted/uninterrupted preflight 必须产生相同 tensor、trajectory 与逻辑状态 hash。sealed corruption、contract drift 或 parity mismatch 为 `INVALID`，不得挑 checkpoint。
8. **域分离 RNG**：新 counter-based contract 以 route hash、training seed、stage、generation、game、decision、simulation、lane、ordinal 派生。两臂共享 arm-neutral initial/chance 根随机条件；action、belief、search 与 self-play batch 分 lane，搜索随机消耗不能移动环境 chance。示范 batch index 流配对，自博弈流按 arm 隔离；PyTorch initialization/learner/device RNG 显式封存，不依赖进程、worker 顺序或 Python global RNG。

本轮仍未授权读取 #44 forgetting-monitor payload。剩余决策是训练探索分布、target 生命周期/代际晋升、monitor 打开时点与用途，以及等墙钟快速试验的预算口径；每代游戏数、更新数、search simulations 和学习率数值继续等待独立吞吐/teacher probe，不从 ZeusAI 或旧 Rapid 常数复制。

**Comment by titanxxh** [comment]
## Grilling round 3 — accepted; decision tree complete

用户选择 Q9–Q12 全部按推荐，冻结最后一层决策：

9. **训练探索**：Search 从完整归一化 root visits 以 `temperature=1.0` 采样，Control 从完整 frozen soft prior 以 `temperature=1.0` 采样；全局不退火，不加入 Dirichlet root noise、epsilon-uniform 或其他额外 exploration。单合法动作直接执行。评测为 deterministic argmax，canonical action ID 最小者打破并列；无 resign、截断或启发式终局。
10. **Target 生命周期与晋升**：一经 trajectory/generation seal，visit、soft-prior 与 terminal-return targets 永不 reanalyse。`g` 的四代近窗产生 `g+1`，`g+1` 通过完整性、有限数值、观测封闭和 manifest gates 后无 strength arena 地顺序晋升；无 parent gate、EMA、best-generation 回选或中途峰值选择。强度只由预注册时点判定，失败为 `NO-GO`，不得回滚挽救。
11. **示范遗忘 monitor**：只有全部数值训练/run 合同冻结且 warm-start/`g0` artifacts 已封存后才首次打开 #44 monitor。只评估 warm-start、`g0` 以及两个 arm 各自预注册的 4h/8h/12h checkpoints，报告 masked NLL、top-1、Age/action-family 分层和相对 warm-start 变化；结果不得停止训练、改 loss/25% anchor、选 checkpoint、调参、判定 Go/No-Go 或支持强度结论。打开后它对本路线永久成为已消费诊断资产，合同改变时不能重新包装成 held-out。
12. **12 小时口径**：Search 与 Control 各自获得 12 小时 active wall-clock，顺序运行，机制试验约 24 小时外加 preflight/评测。空闲主机 throughput probe 先为每臂冻结整数 generation/game/update work packages；Control 可因更便宜而获得更多样本/更新。正式臂执行完整工作包、不按到点截断，4h/8h/12h checkpoint time-matched；实际 active time 必须满足预注册容差，否则为资源/调度 `INVALID`，不能裁数据或改配方。

至此 #43 design tree frontier 为空，用户已确认 shared understanding。每代游戏数、learner updates、warm-start/value-bootstrap budgets、batch size、learning rate、checkpoint cadence、precision、actor count、exact Small dimensions、search simulations/selection depth 和 active-time tolerance 必须由后继 teacher/throughput prototype 在未打开 monitor 与封存评测根前冻结；不得从 ZeusAI 或旧 Rapid 常数静默继承。

**Comment by titanxxh** [comment]
## Resolution

三轮 grilling 的 Q1–Q12 均由用户选择推荐答案，design tree frontier 已为空。核心闭环已冻结为：

- 新 hybrid Transformer 从零初始化；#44 train 只作 policy warm-start/25% Policy 锚；BGA 无 value。
- 冻结完整 policy 决策函数，只以 prior-only 完整扩展合法终局训练 value adapter/head，封存共同 `g0`。
- 从同一 `g0` 分叉 12h Search 与 12h Control：Search 使用 observation-closed IS-PUCT root visits，Control 使用 frozen masked soft prior；两边 value 都只取同代合法终局 actor return。
- 同步零-staleness generations、四代均衡近窗、75/25 self-play/demo、`temperature=1.0`、无 root noise、target 不 reanalyse、checkpoint 无 strength promotion。
- counter-based 域分离 RNG、完整局/generation/checkpoint 原子恢复；合同漂移为 `INVALID`，有效强度门失败为 `NO-GO`。
- #44 monitor 只在数值合同冻结后评估 warm-start、`g0` 和两个臂的 4h/8h/12h snapshots，永不参与训练、调参、选模或 Go/No-Go。

冻结规格：`docs/specs/ai-training/2026-08-27-zeusai-full-extension-training-loop.md`

- file SHA256：`b2d25e7810296558c6902a661f2ed416560f11ed1c699c59a66ee404a96096a6`
- contract：`zeusai-full-extension-training-loop-v1`
- 500 lines / 2,434 words

领域词汇同步在 `CONTEXT.md`，当前 file SHA256 为 `fea6ce55ccc7aa40b7783655625923a5651ada7fbcc25d550f5337811bc9818a`。

Fresh verification：

```text
git diff --check
python3 tools/ai/generate.py --check
python3 -m unittest tools.ai.tests.test_generation tools.ai.tests.test_zeusai_policy_corpus
Ran 9 tests ... OK
```

相对链接检查为零缺失。没有实现 trainer、启动 GPU/CPU 训练、创建 `g0`、读取 forgetting-monitor payload 或读取 Selection/Validation/Test。当前工作树包含用户原有 replay-capture/regularized-pg 修改及本 Wayfinder 的未提交文档/原型/语料冻结文件；本票未 commit、push、merge 或建 worktree。

后继 frontier 是 teacher/throughput prototype 与 12h numerical run contract：必须在 monitor/封存根保持未读时，冻结 exact Small dimensions、warm-start/value-bootstrap budgets、AdamW 数值、simulation/depth、每代 games/updates、并发/precision、checkpoint cadence、RNG algorithm 和 active-time tolerance；不得从 ZeusAI 或旧 Rapid 常数继承。
